source/dol/bases/d_CourseSelectGuide.cpp

```diff
@@ -0,0 +1,1000 @@
+}
+void dCourseSelectGuide_c::executeState_RightDisp() {
+    if (!IsDirectionAllowed(DIR_RIGHT) || mDisableArrows) {
+        mStateMgrRight.changeState(StateID_RightExitAnimeEndCheck);
+    }
+}
+void dCourseSelectGuide_c::finalizeState_RightDisp() {}
+
+void dCourseSelectGuide_c::initializeState_RightExitAnimeEndCheck() {
+    mRightAnim = true;
+    mLayout.AnimeEndSetup(ANIM_LOOP_ARROW_R);
+    mLayout.AnimeStartSetup(ANIM_OUT_ARROW_R, false);
+}
+void dCourseSelectGuide_c::executeState_RightExitAnimeEndCheck() {
+    if (IsDirectionAllowed(DIR_RIGHT)) {
+        mStateMgrRight.changeState(StateID_RightOnStageAnimeEndCheck);
+    } else if (!mLayout.isAnime(ANIM_OUT_ARROW_R)) {
+        mStateMgrRight.changeState(StateID_RightOnStageWait);
+    }
+}
+void dCourseSelectGuide_c::finalizeState_RightExitAnimeEndCheck() {
+    mRightAnim = false;
+}
+
+void dCourseSelectGuide_c::initializeState_ShadowOnStageWait() {}
+void dCourseSelectGuide_c::executeState_ShadowOnStageWait() {
+    if (mShowShadow) {
+        mStateMgrShadow.changeState(StateID_ShadowOnStageAnimeEndCheck);
+    }
+}
+void dCourseSelectGuide_c::finalizeState_ShadowOnStageWait() {
+    mShowShadow = false;
+}
+
+void dCourseSelectGuide_c::initializeState_ShadowOnStageAnimeEndCheck() {
+    mLayout.AnimeStartSetup(ANIM_IN_SHADOW, false);
+}
+void dCourseSelectGuide_c::executeState_ShadowOnStageAnimeEndCheck() {
+    if (!mLayout.isAnime(-1)) {
+        mStateMgrShadow.changeState(StateID_ShadowDisp);
+    }
+}
+void dCourseSelectGuide_c::finalizeState_ShadowOnStageAnimeEndCheck() {}
+
+void dCourseSelectGuide_c::initializeState_ShadowDisp() {}
+void dCourseSelectGuide_c::executeState_ShadowDisp() {
+    if (mHideShadow) {
+        mStateMgrShadow.changeState(StateID_ShadowExitAnimeEndCheck);
+    }
+}
+void dCourseSelectGuide_c::finalizeState_ShadowDisp() {
+    mHideShadow = false;
+}
+
+void dCourseSelectGuide_c::initializeState_ShadowExitAnimeEndCheck() {
+    mLayout.AnimeStartSetup(ANIM_OUT_SHADOW, false);
+}
+void dCourseSelectGuide_c::executeState_ShadowExitAnimeEndCheck() {
+    if (!mLayout.isAnime(-1)) {
+        mStateMgrShadow.changeState(StateID_ShadowOnStageWait);
+    }
+}
+void dCourseSelectGuide_c::finalizeState_ShadowExitAnimeEndCheck() {}
+
+void dCourseSelectGuide_c::initializeState_ScrollGuideOnStageWait() {
+    mScrollGuideRelated = false;
+}
+void dCourseSelectGuide_c::executeState_ScrollGuideOnStageWait() {
+    if (mEndScrollGuide || mGuideRelated) {
+        mEndScrollGuide = false;
+        mScrollGuideRelated = false;
+        mGuideRelated = false;
+        mInMapView = false;
+#if !(defined(VERSION_K) || defined(VERSION_W))
+    } else if (mShowScrollGuide) {
+#else
+    // Don't transition to the next state until the animation is finished.
+    // Otherwise, the "View Map" text would change to "Back to Mario" too early.
+    } else if (mShowScrollGuide && !mLayout.isAnime(ANIM_OUT_UNDER)) {
+#endif
```

Comment:
```suggestion
#if (defined(VERSION_K) || defined(VERSION_W))
    // Don't transition to the next state until the animation is finished.
    // Otherwise, the "View Map" text would change to "Back to Mario" too early.
    } else if (mShowScrollGuide && !mLayout.isAnime(ANIM_OUT_UNDER)) {
#else
    } else if (mShowScrollGuide) {
#endif
```